src/activity.js

```diff
@@ -0,0 +1,55 @@
+const getItemFromLocalStorage = () => {
+  let data = JSON.parse(localStorage.getItem('tasks'));
+  if (!data) {
+    data = [];
+  }
+  return data;
+};
+
+const saveItemToLocalStorage = (tasks) => {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const createTask = () => {
+  const tasks = getItemFromLocalStorage();
+  const newTaskDescription = document.getElementById('to-do-input');
+  const newTask = {
+    description: newTaskDescription.value,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  newTaskDescription.value = '';
+  tasks.push(newTask);
+  saveItemToLocalStorage(tasks);
+};
+
+const updateTask = (tasks, index, value) => {
+  tasks[index].description = value;
+
+  saveItemToLocalStorage(tasks);
+};
+
+const deleteTask = (tasks, index) => {
+  const newTasks = tasks.filter((task, innerIndex) => index !== innerIndex);
+  for (let i = 0; i < newTasks.length; i += 1) {
+    newTasks[i].index = i;
+  }
+  saveItemToLocalStorage(newTasks);
+};
```

Comment:
Well done for making most of the requested changes. However, as stated by the previous reviewer, the changes you made have affected the delete button hence nothing deletes when you click on it. Kindly check and fix it and ensure that the delete and the edit functionalities work very well. :wink: